transformer_engine/common/fused_attn/fused_attn.cpp

```diff
@@ -55,7 +55,7 @@ NVTE_Fused_Attn_Backend nvte_get_fused_attn_backend(
                 || (qkv_layout == NVTE_QKV_Layout::NVTE_KV_INTERLEAVED))) {
       flag_m512 = true;
     }
-    if ((sm_arch_ >= 80)
+    if ((sm_arch_ == 80 || sm_arch_ == 90)
```

Comment:
The arbitrary_seqlen backend supports sm86 and sm89 since cuDNN 8.9.3, so maybe something like this?
if (
#if (CUDNN_VERSION >= 8903)
(sm_arch_ >= 80)
#else
(sm_arch_ == 80 || sm_arch_ == 90)
#end
...

Reply:
Cool, I didn't know that. Let me pend this PR and ask if cuDNN has plan to support Max512 backend for sm86/sm89.